x/bank/client/cli/cli_test.go

```diff
@@ -1,223 +1,289 @@
-		fmt.Sprintf("--gas-prices=%s", sdk.NewDecCoinFromDec(cli.FeeDenom, minGasPrice)), "-y")
-	require.True(t, success)
+	cfg := testutil.DefaultConfig()
+	cfg.NumValidators = 1
 
-	// wait for a block confirmation
-	tests.WaitForNextNBlocksTM(1, f.Port)
+	s.cfg = cfg
+	s.network = testutil.NewTestNetwork(s.T(), cfg)
 
-	f.Cleanup()
+	_, err := s.network.WaitForHeight(1)
+	s.Require().NoError(err)
 }
 
-func TestCLIFeesDeduction(t *testing.T) {
-	t.Parallel()
-	f := cli.InitFixtures(t)
-
-	// start simd server with minimum fees
-	minGasPrice, _ := sdk.NewDecFromStr("0.000006")
-	proc := f.SDStart(fmt.Sprintf("--minimum-gas-prices=%s", sdk.NewDecCoinFromDec(cli.FeeDenom, minGasPrice)))
-	t.Cleanup(func() { proc.Stop(false) })
-
-	// Save key addresses for later use
-	fooAddr := f.KeyAddress(cli.KeyFoo)
-	barAddr := f.KeyAddress(cli.KeyBar)
-
-	fooAmt := testutil.QueryBalances(f, fooAddr).AmountOf(cli.FooDenom)
-
-	// test simulation
-	success, _, _ := testutil.TxSend(
-		f, cli.KeyFoo, barAddr, sdk.NewInt64Coin(cli.FooDenom, 1000),
-		fmt.Sprintf("--fees=%s", sdk.NewInt64Coin(cli.FeeDenom, 2)), "--dry-run")
-	require.True(t, success)
-
-	// Wait for a block
-	tests.WaitForNextNBlocksTM(1, f.Port)
-
-	// ensure state didn't change
-	require.Equal(t, fooAmt.Int64(), testutil.QueryBalances(f, fooAddr).AmountOf(cli.FooDenom).Int64())
-
-	// insufficient funds (coins + fees) tx fails
-	largeCoins := sdk.TokensFromConsensusPower(10000000)
-	success, stdOut, _ := testutil.TxSend(
-		f, cli.KeyFoo, barAddr, sdk.NewCoin(cli.FooDenom, largeCoins),
-		fmt.Sprintf("--fees=%s", sdk.NewInt64Coin(cli.FeeDenom, 2)), "-y")
-	require.Contains(t, stdOut, "insufficient funds")
-	require.True(t, success)
-
-	// Wait for a block
-	tests.WaitForNextNBlocksTM(1, f.Port)
-
-	// ensure state didn't change
-	require.Equal(t, fooAmt.Int64(), testutil.QueryBalances(f, fooAddr).AmountOf(cli.FooDenom).Int64())
-
-	// test success (transfer = coins + fees)
-	success, _, _ = testutil.TxSend(
-		f, cli.KeyFoo, barAddr, sdk.NewInt64Coin(cli.FooDenom, 500),
-		fmt.Sprintf("--fees=%s", sdk.NewInt64Coin(cli.FeeDenom, 2)), "-y")
-	require.True(t, success)
-
-	f.Cleanup()
+func (s *IntegrationTestSuite) TearDownSuite() {
+	s.T().Log("tearing down integration test suite")
+	s.network.Cleanup()
 }
 
-func TestCLIQuerySupply(t *testing.T) {
-	t.Parallel()
-	f := cli.InitFixtures(t)
-
-	// start simd server
-	proc := f.SDStart()
-	t.Cleanup(func() { proc.Stop(false) })
+func (s *IntegrationTestSuite) TestGetBalancesCmd() {
+	buf := new(bytes.Buffer)
+	val := s.network.Validators[0]
+	clientCtx := val.ClientCtx.WithOutput(buf)
+
+	testCases := []struct {
+		name      string
+		args      []string
+		expectErr bool
+		respType  fmt.Stringer
+		expected  fmt.Stringer
+	}{
+		{"no address provided", nil, true, nil, nil},
+		{
+			"total account balance",
+			[]string{
+				val.Address.String(),
+				fmt.Sprintf("--%s=1", flags.FlagHeight),
```

Comment:
It's worth to add a test with only the address as param, with no flags.

Reply:
This brings me another improvement, which I can implement later if you want. 

Now we always set the height flag in these tests, if I remove it they work too (is expected). Maybe we can do a tx and check if setting another height works as expected.

Reply:
The height is set for determinism (i.e. you want to ensure balances). If you don't specify a height, maybe inflation will add more.

Reply:
Good point, thanks!